data/relex-semantic.algs

```diff
@@ -2171,6 +2281,43 @@ COMPARATIVE_Dmum
 =
 <F_L ref degree> = comparative
 
+; This alg is supposed to make a predadj relation for
+; "X is the same as . . ." but it requires a flag for
+; its context that will be really difficult to derive;
+; the real problem is that LG is a mess for comparatives.
+;
+;#TemplateActionAlg
+;PREDADJ_COMPARATIVE
+;<LAB> = \O\.**
+;<F_L ref> = be
+;=
+;<F_R ref links _predadj> += <F_L subj>
+;<F_L subj> = %
+
+; The following alg is for sentences such as . . . 
+; "The book's color is the same as that of the pen."
+; It is necessary to make 'color' the logical object of 'as'
+; and sets a 'compared quality' flag to make it possible
+; to make color the subj of 'of'.  All this is necessary
+; because LG is all wrong here.  There's no link between
+; that+of but there is between same+of.  It's a mess.

```

Comment:
Rather than working around LG bugs, it can be easier to fix LG (maybe).  In this case I see the below:
```
    +------------------------------------Xp------------------------------------+
    +--------------->WV-------------->+                                        |
    +------------Wd-----------+       +---O*m---+--------Mp--------+---Js--+   |
    |      +Ds**c+--YS-+--Dmu-+---Ss--+   +--DD-+--Mp-+--Js--+     |  +Ds**+   |
    |      |     |     |      |       |   |     |     |      |     |  |    |   |
LEFT-WALL the book.n 's.p color.n-u is.v the same.a as.e that.j-p of the pen.n . 

            LEFT-WALL    0.000  Wd+ hWV+ Xp+ 
                  the    0.000  D+ 
               book.n    0.000  Ds**c- YS+ 
                 's.p    0.000  YS- D+ 
            color.n-u    0.000  Dmu- Wd- Ss+ 
                 is.v    0.000  Ss- dWV- O*m+ 
                  the    0.000  DD+ 
               same.a    0.100  DD- O- M+ 
                 as.e    1.000  Mp- J+ 
             that.j-p    2.000  Js- 
                   of    0.000  Mp- J+ 
                  the    0.000  D+ 
                pen.n    0.000  Ds**c- Js- 
                    .    0.000  Xp- RW+ 
```

Observe the costs:  ` as.e    1.000  Mp- J+`  and `that.j-p    2.000  Js-`  which says that LG really really didn't want to make these two links, but it could not find anything better.  So, LG "already knows" its wrong, it just doesn't know what's right.  these sorts of fixes tend to be easier.